pkg/apiserver/webhooks/datavolume-validate.go

```diff
@@ -523,6 +526,28 @@ func (wh *dataVolumeValidatingWebhook) validateDataVolumeSourceSnapshot(snapshot
 	return nil
 }
 
+func validateStorageClassName(spec *cdiv1.DataVolumeSpec, field *k8sfield.Path) *metav1.StatusCause {
+	var sc *string
+
+	if spec.PVC != nil {
+		sc = spec.PVC.StorageClassName
+	} else if spec.Storage != nil {
+		sc = spec.Storage.StorageClassName
+	}
+
+	if sc == nil {
+		return nil
+	}
+	if *sc == "" {
+		return &metav1.StatusCause{
+			Type:    metav1.CauseTypeFieldValueInvalid,
+			Message: fmt.Sprintf("storageClassName is empty"),
```

Comment:
"" storage class name is valid

```yaml
apiVersion: v1
kind: PersistentVolume
metadata:
  name: host-volume
spec:
  capacity:
    storage: 10Gi
  accessModes:
    - ReadWriteOnce
  hostPath:
    path: "/mnt/data"
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: my-pvc
spec:
  volumeMode: Filesystem
  storageClassName: ""
  accessModes:
  - ReadWriteOnce
  resources:
    requests:
      storage: 5Gi
```

```bash
✗ k get pvc
selecting docker as container runtime
NAME     STATUS   VOLUME        CAPACITY   ACCESS MODES   STORAGECLASS   AGE
my-pvc   Bound    host-volume   10Gi       RWO                           3s
```

Reply:
So had this discussion with Arnon, if you read [this](https://kubernetes.io/blog/2023/01/05/retroactive-default-storage-class/), it does say "" is equivalent to missing, due to historical reasons, but having a blank string is sort of gray at this point. This is fixing a reported issue where someone is passing a blank storage class by accident. I think we should disallow blank, and if someone wants to use the default, they can omit the field.

Reply:
I dunno, this pr provides no way to explicitly bind to "" storageclass .  That combined with the fact that in 1.26 "" on PVC will bind to default storage class if exists, makes me believe that we should leave things as-is OR allow updating the storageclass on the DV as you can do with PVCs in 1.26

Reply:
Finally tried this out on a 1.26 cluster.  Apparently, I misread the doc.  "" will not bind to the default storageclass.  And pvc with "" storageClassName can't be updated.

Nevertheless because "" is a legit value I don't think we should disallow it